core/src/main/java/com/linecorp/armeria/common/stream/AbstractStreamMessageDuplicator.java

```diff
@@ -256,13 +323,22 @@ boolean isDuplicable() {
             return state == State.DUPLICABLE;
         }
 
+        boolean isLastDownstreamAdded() {
+            return state == State.LAST_DOWNSTREAM_ADDED;
+        }
+
         void close() {
-            if (stateUpdater.compareAndSet(this, State.DUPLICABLE, State.CLOSED)) {
+            if (setState(State.DUPLICABLE, State.CLOSED)) {
```

Comment:
Just curious. We do not call `upstream.abort()` and `cleanup()` when the state is `LAST_DOWNSTREAM_ADDED`, right?